zipline-kotlin-plugin/src/main/kotlin/app/cash/zipline/ktbridge/plugin/BridgedInterface.kt

```diff
@@ -17,28 +17,112 @@ package app.cash.zipline.ktbridge.plugin
 
 import org.jetbrains.kotlin.backend.common.extensions.IrPluginContext
 import org.jetbrains.kotlin.ir.IrElement
+import org.jetbrains.kotlin.ir.builders.IrBuilderWithScope
+import org.jetbrains.kotlin.ir.builders.irCall
+import org.jetbrains.kotlin.ir.builders.irExprBody
+import org.jetbrains.kotlin.ir.builders.irGet
+import org.jetbrains.kotlin.ir.declarations.IrClass
+import org.jetbrains.kotlin.ir.declarations.IrProperty
+import org.jetbrains.kotlin.ir.declarations.IrValueParameter
+import org.jetbrains.kotlin.ir.expressions.IrFunctionAccessExpression
 import org.jetbrains.kotlin.ir.symbols.IrClassSymbol
+import org.jetbrains.kotlin.ir.symbols.IrSimpleFunctionSymbol
 import org.jetbrains.kotlin.ir.types.IrSimpleType
 import org.jetbrains.kotlin.ir.types.IrType
 import org.jetbrains.kotlin.ir.types.classFqName
+import org.jetbrains.kotlin.ir.types.typeWith
+import org.jetbrains.kotlin.ir.util.functions
 import org.jetbrains.kotlin.ir.util.isInterface
+import org.jetbrains.kotlin.ir.util.properties
 import org.jetbrains.kotlin.ir.util.substitute
 import org.jetbrains.kotlin.name.FqName
+import org.jetbrains.kotlin.name.Name
 
 /**
  * A user-defined interface (like `EchoService` or `Callback<String>`) and support for either
  * implementing it ([KtBridgeGetRewriter]) or calling it ([KtBridgeSetRewriter]).
  *
  * This class tracks the interface type (like `EchoService` or `Callback<String>`) and its
  * implementation class (that doesn't know its generic parameters).
+ *
+ * This class can declare a `KSerializer` property for each unique parameter type and return type on
+ * the interface. Use [declareSerializerProperties] to create these, then [serializerExpression] to
+ * access a specific serializer. Declaring serializers is useful to fast fail if ever a serializer
+ * is required but not configured.
  */
 internal class BridgedInterface(
+  private val pluginContext: IrPluginContext,
+  private val ktBridgeApis: KtBridgeApis,
+
   /** A specific type identifier that knows the values of its generic parameters. */
   val type: IrType,
 
   /** A potentially-generic declaration that doesn't have values for its generic parameters. */
-  val classSymbol: IrClassSymbol,
+  private val classSymbol: IrClassSymbol,
```

Comment:
Were these made non-private for dev / testing only?

Reply:
This was made private ’cause it’s no longer needed elsewhere.